src/sentry/static/sentry/app/views/settings/components/forms/richListField.tsx

```diff
@@ -194,31 +212,31 @@ class RichList extends React.PureComponent {
   }
 }
 
-export default class RichListField extends React.PureComponent {
-  static propTypes = {
-    // TODO(ts)
-    // ...InputField.propTypes,
-    ...RichListProps,
-  };
-
-  renderRichList = fieldProps => {
-    const richListProps = getDefinedProps(RichListProps, this.props);
-    const {value, ...props} = fieldProps;
-
-    // We must not render this field until `setValue` has been applied by the
-    // model, which is done after the field is mounted for the first time. To
-    // check this, we cannot use Array.isArray because the value passed in by
-    // the model might actually be an ObservableArray.
-    if (typeof value === 'string' || value.length === undefined) {
-      return null;
-    }
-
-    return <RichList {...props} value={[...value]} {...richListProps} />;
-  };
-
-  render() {
-    return <InputField {...this.props} field={this.renderRichList} />;
-  }
+/**
+ * A 'rich' dropdown that provides action hooks for when item
+ * are selected/created/removed.
+ *
+ * An example usage is the debug image selector where each 'source' option
+ * requires additional configuration data.
+ */
+export default function RichListField(props: RichListProps & InputField['props']) {
+  return (
+    <InputField
+      {...props}
+      field={(fieldProps: RichListProps) => {
+        const {value, ...otherProps} = fieldProps;
+
+        // We must not render this field until `setValue` has been applied by the
+        // model, which is done after the field is mounted for the first time. To
+        // check this, we cannot use Array.isArray because the value passed in by
+        // the model might actually be an ObservableArray.
+        if (typeof value === 'string' || value.length === undefined) {
```

Comment:
shouldn't we add a `?` in value?
```suggestion
        if (typeof value === 'string' || value?.length === undefined) {
```